jacodb-analysis/src/main/kotlin/org/jacodb/analysis/ifds2/Edge.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ *  Copyright 2022 UnitTestBot contributors (utbot.org)
+ * <p>
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ * <p>
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.jacodb.analysis.ifds2
+
+import org.jacodb.analysis.engine.IfdsEdge
+import org.jacodb.analysis.ifds2.taint.TaintFact
+import org.jacodb.api.JcMethod
+
+data class Edge<out Fact>(
+    val from: Vertex<Fact>,
+    val to: Vertex<Fact>,
+) {
+    init {
+        require(from.method == to.method)
+    }
+
+    val method: JcMethod
+        get() = from.method
+
+    companion object {
+        // constructor
+        operator fun invoke(edge: IfdsEdge): Edge<TaintFact> {
+            return Edge(Vertex(edge.from), Vertex(edge.to))
+        }
+    }
+}
+
+fun Edge<TaintFact>.toIfds(): IfdsEdge = IfdsEdge(from.toIfds(), to.toIfds())
```

Comment:
If we are going to completely remove "old ifds", this converters can be safely removed

Reply:
Ok, let's remove all such stuff in this PR

Reply:
Okay, I have removed these "toIfds" methods since they are unsed. However, regarding the complete removal of all "old ifds code" -- I still think it should be done in a separate PR in order for us not to lose everything that might be useful here, and to just move forward with other tasks.